tools/predict_net.py

```diff
@@ -0,0 +1,207 @@
+from time import time
+
+import numpy as np
+import pandas as pd
+import cv2
+import torch
+
+import slowfast.utils.checkpoint as cu
+import slowfast.utils.distributed as du
+import slowfast.utils.logging as logging
+import slowfast.utils.misc as misc
+from slowfast.datasets import loader
+from slowfast.models import model_builder
+# from slowfast.utils.meters import TestMeter
+
+logger = logging.get_logger(__name__)
+
+def imresize(im, dsize):
+    '''
+    Resize the image to the specified square sizes and 
+    maintain the original aspect ratio using padding.
+
+    Args:
+
+        im -- input image.
+        dsize -- output sizes, can be an integer or a tuple.
+
+    Returns:
+
+        resized image.
+    '''
+    if type(dsize) is int:
+        dsize = (dsize, dsize)
+        
+    im_h, im_w, _ = im.shape
+    to_w, to_h = dsize
+    scale_ratio = min(to_w/im_w, to_h/im_h)
+    new_im = cv2.resize(im,(0, 0), 
+                        fx=scale_ratio, fy=scale_ratio, 
+                        interpolation=cv2.INTER_AREA)
+    new_h, new_w, _ = new_im.shape
+    padded_im = np.full((to_h, to_w, 3), 128)
+    x1 = (to_w-new_w)//2
+    x2 = x1 + new_w
+    y1 = (to_h-new_h)//2
+    y2 = y1 + new_h
+    padded_im[y1:y2, x1:x2, :] = new_im 
+    
+    return padded_im
+
+class VideoReader(object):
+
+    def __init__(self, source):
+        self.source = source
+        try:  # OpenCV needs int to read from webcam
+            self.source = int(source)
+        except ValueError:
+            pass
+
+    def __iter__(self):
+        self.cap = cv2.VideoCapture(self.source)
+        # self.cap.set(cv2.CAP_PROP_FRAME_WIDTH, 1280)
+        # self.cap.set(cv2.CAP_PROP_FRAME_HEIGHT, 720)
+        if not self.cap.isOpened():
+            raise IOError('Video {} cannot be opened'.format(self.source))
+        return self
+
+    def __next__(self):
+        was_read, frame = self.cap.read()
+        if not was_read:
+            # raise StopIteration
+            ## reiterate the video instead of quiting.
+            self.cap.set(cv2.CAP_PROP_POS_FRAMES, 0)
+            frame = None
+
+        return was_read, frame
+
+    def clean(self):
+        self.cap.release()
+        cv2.destroyAllWindows()
+
+
+def predict(cfg):
+    # Set random seed from configs.
+    np.random.seed(cfg.RNG_SEED)
+    torch.manual_seed(cfg.RNG_SEED)
+
+    # Setup logging format.
+    logging.setup_logging()
+
+    # Print config.
+    logger.info("Predict with config:")
+    logger.info(cfg)
+    # Build the video model and print model statistics.
+    model = model_builder.build_model(cfg)
+    model.eval()
+    misc.log_model_info(model)
+
+   # Load a checkpoint to test if applicable.
+    if cfg.TEST.CHECKPOINT_FILE_PATH != "":
+        ckpt = cfg.TEST.CHECKPOINT_FILE_PATH
+        convert_from_caffe2 = cfg.TEST.CHECKPOINT_TYPE == "caffe2"
+    elif cu.has_checkpoint(cfg.OUTPUT_DIR):
+        ckpt = cfg.OUTPUT_DIR
```

Comment:
Could you share me the reason that you are not preferring of using 
`last_checkpoint = cu.get_last_checkpoint(cfg.OUTPUT_DIR)`
In `train_net.py` or `'test_net;.py`? I am happy to modify the api to make it easier to use for your use case.

Reply:
I didn't look carefully into `cu` and hence didn't know there would be a `checkpoints/` in `cfg.OUTPUT_DIR` XP.  I'm going to change `ckpt = cfg.OUTPUT_DIR` to `ckpt = cu.get_last_checkpoint(cfg.OUTPUT_DIR)`.  It seem
 `cu.get_last_checkpoint(...)` does similar things to `cu.has_checkpoint(...)` in the beginning, if there could be a way to reuse the function or merging them together would help to keep the code clean, yet it's not that big deal.